application/src/objects/obstacles/obstacle-factory.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var _          = require('underscore');
+var ClownNose  = require('./clown-nose');
+var DynamicBox = require('./dynamic-box');
+var StaticBox  = require('./static-box');
+
+var ObstacleFactory = function(state) {
+    this.state      = state;
+};
+
+ObstacleFactory.prototype.types = {
+    'ClownNose'  : ClownNose,
+    'DynamicBox' : DynamicBox,
+    'StaticBox'  : StaticBox
+};
+
+ObstacleFactory.prototype.loadAssets = function(types)
+{
+    _.each(types, function(type) {
+        if (this.types[type]) {
+            this.state.load.image(type, this.types[type].prototype.getSpritePath());
+        } else {
+            throw new Error('Attempted to load assets for unknown class ' + type);
+        }        
+    }, this);
+};
+
+ObstacleFactory.prototype.getNew = function(type, x, y, collisionGroup)
```

Comment:
The only reason I hesitate to say `getSprite` is because to me, the sprite is just the 2D image drawn on the screen, but the object classes we're creating also include specifications for the physical body and the update functions handle more than simply animating the sprite. And with the way the `type` key (which I envision will almost always be the class name) is the first argument, calls to this function read like `context.obstacleFactory.getNew('ClownNose', [other arguments...])`, which I kinda like.